tests/integration/targets/utils_keep_keys/tasks/simple.yaml

```diff
@@ -94,3 +94,13 @@
   ansible.builtin.assert:
     that:
       - keep_tomcat['tomcat'] == result['msg']
+
```

Comment:
@roverflow can we make the input set more complicated to see how it behaves
have a dict in dict / list in dict / string in dict / string in list

Example
```
- name: Setup data for multiple keys in dict
  ansible.builtin.set_fact:
    tomcat_data:
      tomcat:
        tomcat1:
          name: tomcat1
        tomcat2:
          name: tomcat2
          tomcat3:
            name: tomcat3
        tomcat3:
          name: tomcat3
        tomcat4:
          name: tomcat3
          tomcat3:
            tomcat3:
              name: tomcattest
            checktomcat3: stringinput
      tomcats_block:
        - tomcat1
        - tomcat2
```

Reply:
got it, and what are the keys that we want to keep for this input set?

Reply:
added more tests!